src/test/fuzz/script.cpp

```diff
@@ -56,46 +56,8 @@ FUZZ_TARGET_INIT(script, initialize_script)
         assert(script == decompressed_script);
     }
 
-    CTxDestination address;
-    TxoutType type_ret;
-    std::vector<CTxDestination> addresses;
-    int required_ret;
-    bool extract_destinations_ret = ExtractDestinations(script, type_ret, addresses, required_ret);
-    bool extract_destination_ret = ExtractDestination(script, address);
-    if (!extract_destinations_ret) {
-        assert(!extract_destination_ret);
-        if (type_ret == TxoutType::MULTISIG) {
-            assert(addresses.empty() && required_ret == 0);
-        } else {
-            assert(type_ret == TxoutType::PUBKEY ||
-                   type_ret == TxoutType::NONSTANDARD ||
-                   type_ret == TxoutType::NULL_DATA);
-        }
-    } else {
-        assert(required_ret >= 1 && required_ret <= 16);
-        assert((unsigned long)required_ret == addresses.size());
```

Comment:
This assertion fails:

https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39152

It might be good to merge this pull request or apply the changes to this fuzz test early.

Reply:
I'm looking into this now to see why this assertion failed. These tests were added in https://github.com/bitcoin/bitcoin/pull/20772. I'm also curious why we're only seeing this now but haven't had the chance to look into this and know what's going on

Reply:
I just went ahead and fixed that assertion as I believe it was a buggy assertion. I didn't want this PR to get rushed bc of this, and figured we may want this for correctness. #23091  -- I realize the PR is kinda pointless, so if its best to close it and this is ready to merge, just lmk